src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -332,13 +334,22 @@ public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort();
+                MoveItemOnFirstPlace(PSVersionInfo.PSEditionName, keyList);
+                MoveItemOnFirstPlace(PSVersionInfo.PSVersionName, keyList);
+                return keyList.ToArray();
```

Comment:
`PSVersionTable` doesn't need to be read-only for the `Keys` to be read-only. `Hashtable.Keys` returns an object of `System.Collections.Hashtable+KeyCollection`, which expose no public members for you to mutate it.
This would be a pure optimization and is not required for this fix.